daemon/cmd/endpoint_restore.go

```diff
@@ -131,6 +132,80 @@ func (r *endpointRestorer) checkLink(linkName string) error {
 	return err
 }
 
+// endpointLinkType returns the type of network link used by the endpoint.
+func endpointLinkType(ep *endpoint.Endpoint) (string, error) {
+	if ep.IsProperty(endpoint.PropertyFakeEndpoint) {
+		return "", nil
+	}
```

Comment:
suggestion: let's move this check into the caller, along side the [check for pod endpoints](https://github.com/cilium/cilium/pull/42482/files#diff-31f015a7266e62c996a65e83e44098951cf4652e4ecfc4c83f6aba1ab9bd539eR169). I think it's better suited there.